dotCMS/src/main/java/com/dotcms/dotpubsub/QueuingPubSubWrapper.java

```diff
@@ -141,4 +190,126 @@ public DotPubSubEvent lastEventOut() {
+                    iterator.remove();
+                }
+
+                outgoingMessages.clear();
+            } finally {
+                if (outgoingMessages != null) {
+                    try {
+                        pool.returnObject(outgoingMessages);
+                    } catch (Exception e) {
+                        Logger.error(QueuingPubSubWrapper.class, e.getMessage());
+
+                        try {
+                            pool.invalidateObject(outgoingMessages);
+                        } catch (Exception exception) {
+                            Logger.error(QueuingPubSubWrapper.class, e.getMessage());
+                        }
+                    }
+                }
+            }
+
+            Logger.debug(this.getClass(), () -> "Running QueuingPubSubWrapper.publishQueue Finish Thread" + Thread.currentThread().getName());
+
+        }
+    }
+
+    private static class CollectionWrapper<E> implements Collection<E> {
+
+        private final Collection<E> rootCollection;
+
+        public CollectionWrapper(final Collection<E> rootCollection) {
+            this.rootCollection = rootCollection;
+        }
+
+        @Override
+        public Iterator<E> iterator() {
+            return rootCollection.iterator();
+        }
+
+        @NotNull
+        @Override
+        public Object[] toArray() {
+            return rootCollection.toArray();
+        }
+
+        @NotNull
+        @Override
+        public <T> T[] toArray(T[] array) {
+            return rootCollection.toArray(array);
+        }
+
+        @Override
+        public boolean add(E element) {
+            return rootCollection.add(element);
+        }
+
+        @Override
+        public boolean remove(Object element) {
+            return rootCollection.remove(element);
+        }
+
+        @Override
+        public boolean containsAll(Collection<?> collection) {
+            return rootCollection.containsAll(collection);
+        }
+
+        @Override
+        public boolean addAll(Collection<? extends E> collection) {
+            return rootCollection.addAll(collection);
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> collection) {
+            return rootCollection.removeAll(collection);
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> collection) {
+            return rootCollection.retainAll(collection);
+        }
+
+        @Override
+        public void clear() {
+            rootCollection.clear();
+        }
+
+        @Override
+        public int size() {
+            return rootCollection.size();
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return false;
+        }
+
+        @Override
+        public boolean contains(final Object element) {
+            return rootCollection.contains(element);
+        }
+
+        @Override
+        public boolean equals(final Object another){
```

Comment:
Is this OK?

Reply:
Yes, really this is not necessary because it is the default Object's implementation but I am created a Collection Wrapper class just to not use the default equals method implementation in AbstractCollection so I think is good idea have the equals implementation explicitly there

Reply:
Ok.